metals/src/main/scala/scala/meta/internal/metals/FormattingProvider.scala

```diff
@@ -0,0 +1,91 @@
+package scala.meta.internal.metals
+
+import scala.meta._
+import org.eclipse.{lsp4j => l}
+import MetalsEnrichments._
+import scala.language.reflectiveCalls
+import java.util
+import java.nio.file.Files
+import scala.util.Try
+import scala.meta.internal.tokenizers.PlatformTokenizerCache
+import com.typesafe.config.ConfigFactory
+
+/**
+ * Implement text formatting using Scalafmt
+ */
+final class FormattingProvider(
+    workspace: AbsolutePath,
+    buffers: Buffers,
+    embedded: Embedded,
+    userConfig: () => UserConfiguration
+) {
+
+  def format(path: AbsolutePath): util.List[l.TextEdit] = {
+    val input = path.toInputFromBuffers(buffers)
+    val fullDocumentRange = Position.Range(input, 0, input.chars.length).toLSP
+    val defaultScalafmtVersion = "1.5.1"
+    (for {
+      scalafmt <- scalafmtConfigPath match {
+        case Some(configPath) =>
+          val scalafmtConf = ConfigFactory.parseFile(configPath.toNIO.toFile)
+          val scalafmtVersion = Try(scalafmtConf.getString("version")).toOption
+            .getOrElse(defaultScalafmtVersion)
+          classloadScalafmt(scalafmtVersion)
+        case _ =>
+          scribe.info(
```

Comment:
```scala
scribe.error("format error: no .scalafmt.conf file (To fix this problem create an empty file $abspath)")
```
We should also do a `showMessageRequest` with text

> Unable to format because there is no `.scalafmt.conf` file in this workspace. To fix this problem create an empty file in $path.

We can add a button "Create .scalafmt.conf" that writes to that file.